cli/src/cmd/forge/selectors.rs

```diff
@@ -89,6 +113,63 @@ impl SelectorsSubcommands {
             }
+            SelectorsSubcommands::Collision { mut first_contract, mut second_contract, build } => {
+                // Build first project
+                let first_project = build.project()?;
+                let first_outcome = if let Some(ref mut contract_path) = first_contract.path {
+                    let target_path = canonicalize(&*contract_path)?;
+                    *contract_path = target_path.to_string_lossy().to_string();
+                    compile::compile_files(&first_project, vec![target_path], true)
+                } else {
+                    compile::suppress_compile(&first_project)
+                }?;
+
+                // Build second project
+                let second_project = build.project()?;
+                let second_outcome = if let Some(ref mut contract_path) = second_contract.path {
+                    let target_path = canonicalize(&*contract_path)?;
+                    *contract_path = target_path.to_string_lossy().to_string();
+                    compile::compile_files(&second_project, vec![target_path], true)
+                } else {
+                    compile::suppress_compile(&second_project)
+                }?;
+
+                // Find the artifacts
+                let first_found_artifact = first_outcome.find_contract(&first_contract);
+                let second_found_artifact = second_outcome.find_contract(&second_contract);
+
+                // Unwrap inner artifacts
+                let first_artifact = first_found_artifact.ok_or_else(|| {
+                    eyre::eyre!("Failed to extract first artifact bytecode as a string")
+                })?;
+                let second_artifact = second_found_artifact.ok_or_else(|| {
+                    eyre::eyre!("Failed to extract second artifact bytecode as a string")
+                })?;
+
+                // Check method selectors for collisions
+                let first_method_map = first_artifact.method_identifiers.as_ref().unwrap();
+                let second_method_map = second_artifact.method_identifiers.as_ref().unwrap();
+
+                let mut colliding_methods = Vec::new();
+                for (k1, v1) in first_method_map {
+                    if let Some(k2) =
+                        second_method_map
+                            .iter()
+                            .find_map(|(k2, v2)| if v1 == v2 { Some(k2) } else { None })
+                    {
+                        colliding_methods.push((k1, k2));
+                    }
+                }
+
+                if colliding_methods.is_empty() {
+                    println!("No colliding method selectors between the two contracts.");
+                } else {
+                    colliding_methods.iter().for_each(|t| {
+                        println!("\"{}\": {:?}", first_method_map.get(t.0).unwrap(), t)
+                    });
+                    return Err(eyre::eyre!("At least one collision was found"))
+                }
```

Comment:
Hmm, I was thinking, maybe this command doesn't need to error on here? We can color the output with `Paint::red` for saying we found collisions, and then maybe pretty print on a table instead. wdyt @mds1 / @mattsse ?

Reply:
I think a common use of this command is to run it in CI to make sure you don't merge anything with selector collisions, so my suggestion would be to either:
1. Have this command always exit with an error (i.e. `std::process::exit(1)`)` if a collision is found, OR
2. Only exit with an error if a `--check` flag is added

Currently we do both depending on the command,e.g. `forge build --sizes` exits with an error if anything is over the size limit, but `forge fmt` needs a `--check` flag. I can see an argument that the former should also have a `--check` flag, so I'm indifferent here and maybe we start requiring the `--check` flag as a standard?

Also, I'm now realizing this may all be tangential to returning an `Err` from this function 😅

Reply:
I'm more lenient towards adding the `--check` flag so that we can support both of your guys' visions.
Regarding the pretty print on a table, yesterday I was looking at [prettytable-rs](https://github.com/phsym/prettytable-rs) but idk what you guys think about adding it as a dep.

Reply:
Other parts of the code use `comfy-table` to pretty-print stuff, that's what `forge build --sizes` uses, I think `forge inspect` too